presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -1611,4 +1617,10 @@ private static Double mapL2Norm(Block map)
 
         return Math.sqrt(norm);
     }
+
+    private static BigDecimal bigDecimalRound(double num, long decimals)
+    {
+        return BigDecimal.valueOf(num)
```

Comment:
On the profile BigDecimal.valueOf used Double.toString which was expensive and almost 60% of the regression. BigDecimal.setScale was the next one, but I believe setScale can't be avoided.

Would it be better to use new BigDecimal(num, MathContext). 

We can have a cache of MathContext for decimals less than 20 to avoid repeated allocation of the MathContext.

Reply:
Thanks for the feedback @arunthirupathi ! I just updated the branch to use `new BigDecimal(num, MathContext.DECIMAL128)` instead which will indeed prevent from using toString() under the hood.
I also added a line to cap `decimals` to `Decimals.MAX_PRECISION`  (= 38) in round(), since having a higher precision does not make much sense for doubles and floats.
With these changes, I'm getting similar performance than the current round() implementation in the [benchmark I wrote](https://gist.github.com/sviscaino/945ca6c1b63762bdd082bf512a184787), even when forcing ROUND() to land on an edge-case (e.g. `ROUND((totalprice * i)*(cast(1e100 as double)), 1000)`).
Also, the bigdecimal approach will only be called if num*10**decimals overflows a long - which is a very corner edge case.